pllug-presentation-system/presentation-data/presentationelementfactory.h

```diff
@@ -0,0 +1,15 @@
+#ifndef PRESENTATIONELEMENTFACTORY_H
+#define PRESENTATIONELEMENTFACTORY_H
+
+#include <memory>
+#include <QString>
+
+#include "presentationelement.h"
```

Comment:
It is better to use forward declaration here instead of include "presentationelement.h", as we have a pointer to PresentationElement.